### ISSUE TITLE
Knife cookbook list seems to present a hash, but actually outputs an array of strings.

### ISSUE THREAD

**Issue by scottvidmar** [chef/chef#10404]
## Description
The format of the knife-tidy reports aren't the same as the knife cookbook list --all -F json. While knife tidy presents an easily traversed JSON structure, the knife cookbook list spits out an array of strings in JSON format that requires additional parsing to pull versions out, like so:

My example:
```
$ knife cookbook list --all -F json
[
  "chef-client    11.5.0",
  "cron           6.2.2",
  "logrotate      2.2.2",
  "starter        1.0.0",
  "updater-test   0.1.0"
]
```

While technically correct, this doesn't seem like it's intended. Seems like this happens in this method: https://github.com/chef/chef/blob/2bfc9106cfa6a3df12204bd255b5f7007c2f8f30/lib/chef/knife/core/generic_presenter.rb#L208

## Chef Version
```
$ chef -v
Chef Workstation version: 20.9.136
Chef Infra Client version: 16.4.41
Chef InSpec version: 4.22.22
Chef CLI version: 3.0.27
Chef Habitat version: 1.6.56
Test Kitchen version: 2.6.0
Cookstyle version: 6.16.10
```

## Platform Version
Running on MacOS 10.14.6

## Client Output
```
$ knife cookbook list --all -F json
[
  "chef-client    11.5.0",
  "cron           6.2.2",
  "logrotate      2.2.2",
  "starter        1.0.0",
  "updater-test   0.1.0"
]
```

**Comment by lamont-granquist** [comment]
This probably was intended though from some logic 10 years ago, and has been a stable API for as long as Chef has been around.

There's almost certainly code out there were someone has piped this output through sed to extract the values people want which would be broken by changing it.

**Comment by tas50** [comment]
Even though this is a 10 year old API it's terrible and we should be ok with fixing it. This output is next to useless.

**Comment by tpowell-progress** [comment]
Addressed in 18.3.0 release / #13654 / v18.2.18
